compiler/resolution/lowerIterators.cpp

```diff
@@ -139,6 +139,24 @@ static void nonLeaderParCheck()
   USR_STOP();
 }
 
+static bool isVirtualIterator(Symbol* iterator) {
+  Vec<Type*> children = iterator->type->dispatchChildren;
+  return !((children.n == 0) || (children.n == 1 && children.v[0] == dtObject));
```

Comment:
How would dtObject ever be a child of of an iterator type?

Reply:
I see that similar checks have been around for a long time, but it seems very strange.

Reply:
I was similarly confused by this, but I wanted to limit the scope of this change since we're passed feature freeze. I have an untested branch that just converts this to `return children.n > 0`, which is also what we do in other places in the compiler including later in `expandForLoop()` and in `handlePolymorphicIterators()`

Any objections to sticking with what was there for now?

Reply:
No objection.